packages/core/src/api/nodeConversions/blockToNode.ts

```diff
@@ -322,16 +320,10 @@ function blockOrInlineContentToContentNode(
  * Converts a BlockNote block to a Prosemirror node.
  */
 export function blockToNode(
-  block: PartialBlock<any, any, any>,
+  block: Block<any, any, any>,
   schema: Schema,
   styleSchema: StyleSchema = getStyleSchema(schema),
 ) {
-  let id = block.id;
-
-  if (id === undefined) {
-    id = UniqueID.options.generateID();
-  }
-
```

Comment:
I'm actually ok with this function generating ids if the block doesn't have one. I think this makes the caller's life more difficult to be forced to present an ID